pkg/controller/manager.go

```diff
@@ -63,6 +63,7 @@ func (m *Manager) UpdateController(name string, params ControllerParams) *Contro
 
 	if oldCtrl, ok := m.controllers[name]; ok {
 		oldCtrl.stopController()
+		m.removeController(oldCtrl)
```

Comment:
Is this a bugfix that should be backported?

Reply:
The real bug here is that a new uuid is generated every time the controller is updated which is horribly wrong. Let's fix it properly outside of this PR.

Reply:
I've filed https://github.com/cilium/cilium/issues/4066

Reply:
Merged separately.